extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceBuildTimeConfig.java

```diff
@@ -32,5 +31,5 @@ public static final FlywayDataSourceBuildTimeConfig defaultConfig() {
      * scanned recursively down non-hidden directories.
      */
     @ConfigItem(defaultValue = DEFAULT_LOCATION)
-    public List<String> locations;
+    public Optional<String[]> locations;
```

Comment:
This one shouldn't change to an array (almost every usage of this property treats it as a list).  And, it wasn't `Optional` before so it should not become `Optional` now.